src/tools/kata-ctl/src/check.rs

```diff
@@ -197,6 +212,81 @@ pub fn check_official_releases() -> Result<()> {
     Ok(())
 }
 
+#[cfg(any(target_arch = "x86_64"))]
+pub fn check_kernel_module_loaded(module: &str, parameter: &str) -> Result<String, String> {
+    const MODPROBE_PARAMETERS_DRY_RUN: &str = "--dry-run";
+    const MODPROBE_PARAMETERS_FIRST_TIME: &str = "--first-time";
+    const MODULES_PATH: &str = "/sys/module";
+
+    // Partial check w/ modinfo
+    let output_modinfo = Command::new(MODINFO_PATH)
+        .arg(module)
+        .output()
+        .expect("failed to execute process");
+
+    let status_modinfo = output_modinfo.status;
+
+    // Partial check w/ modprobe
+    let output_modprobe = Command::new(MODPROBE_PATH)
+        .arg(MODPROBE_PARAMETERS_DRY_RUN)
+        .arg(MODPROBE_PARAMETERS_FIRST_TIME)
+        .arg(module)
+        .stdout(Stdio::piped())
+        .output()
+        .expect("failed to execute process");
+
+    let status_modprobe = output_modprobe.status;
+
+    // Return module parameter value on success
+    if status_modinfo.success() && !status_modprobe.success() {
+        // Get the parameter value
+        let module_path = format!("{}/{}/parameters/{}", MODULES_PATH, module, parameter);
+
+        match get_single_cpu_info(&module_path, parameter) {
```

Comment:
I am confused, why are you using `get_single_cpu_info` here ?

Reply:
Basically `get_single_cpu_info` reads a file passed as an argument and looks for a string within that file.
I used this function to read the parameter value of the module.

Reply:
I suggest you use a separate utility function to do so, you could refactor a bit to make `get_single_cpu_info` to make use of it if you want.

Reply:
I agree with @amshinde.

Reply:
@amshinde, it was fixed by renaming the function `get_cpu_info()` to `read_file_contents()`